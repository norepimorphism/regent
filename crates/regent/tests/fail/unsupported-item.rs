fn main() {
    // Module.
    #[regent::bitwise]
    mod module {}
    // `use` declaration.
    #[regent::bitwise]
    use module::*;
    // Function definition.
    #[regent::bitwise]
    fn function() {}
    // Type definition.
    #[regent::bitwise]
    type Type = ();
    // Union definition.
    #[regent::bitwise]
    union Union {}
    // Constant item.
    #[regent::bitwise]
    const CONST: () = ();
    // Static item.
    #[regent::bitwise]
    static STATIC: () = ();
    // Trait definition.
    #[regent::bitwise]
    trait Trait {}
    // Implementation.
    #[regent::bitwise]
    impl Trait for Union {}
    // Extern block.
    #[regent::bitwise]
    extern "C" {}
}
