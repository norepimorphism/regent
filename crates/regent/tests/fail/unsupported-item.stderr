error: item must be a struct or enum
 --> tests/fail/unsupported-item.rs:4:5
  |
4 |     mod module {}
  |     ^^^^^^^^^^^^^

error: item must be a struct or enum
 --> tests/fail/unsupported-item.rs:7:5
  |
7 |     use module::*;
  |     ^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:10:5
   |
10 |     fn function() {}
   |     ^^^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:13:5
   |
13 |     type Type = ();
   |     ^^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:16:5
   |
16 |     union Union {}
   |     ^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:19:5
   |
19 |     const CONST: () = ();
   |     ^^^^^^^^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:22:5
   |
22 |     static STATIC: () = ();
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:25:5
   |
25 |     trait Trait {}
   |     ^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:28:5
   |
28 |     impl Trait for Union {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error: item must be a struct or enum
  --> tests/fail/unsupported-item.rs:31:5
   |
31 |     extern "C" {}
   |     ^^^^^^^^^^^^^
